h/api/schemas.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+"""Classes for validating data passed to the annotations API."""
+
+import jsonschema
+from jsonschema.exceptions import best_match
+
+
+# These annotation fields are not to be set by the user.
+PROTECTED_FIELDS = ['created', 'updated', 'user', 'id']
+
+
+class ValidationError(Exception):
+    pass
+
+
+class JSONSchema(object):
+
+    """
+    Validate data according to a Draft 4 JSON Schema.
+
+    Inherit from this class and override the `schema` class property with a
+    valid JSON schema.
+    """
+
+    schema = {}
+
+    def __init__(self):
+        self.validator = jsonschema.Draft4Validator(self.schema)
+
+    def validate(self, data):
+        """
+        Validate `data` according to the current schema.
+
+        :param data: The data to be validated
+        :return: valid data
```

Comment:
My intent was that validate should be referentially transparent, so I suppose it should make a copy and return that.